common/src/main/java/org/apache/cassandra/sidecar/common/StorageOperations.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.sidecar.common;
+
+import java.util.Map;
+
+/**
+ * An interface that defines interactions with the storage system in Cassandra
+ */
+public interface StorageOperations
```

Comment:
nit: should it be called `JmxOperations`, which interacts with different components in cassandra (not limited to storage) via JMX, or `StorageJmxOperations` if want to limit for storage service.

Reply:
I implemented this following the comment in this class: https://github.com/apache/cassandra-sidecar/blob/trunk/common/src/main/java/org/apache/cassandra/sidecar/common/ICassandraAdapter.java#L26 . This makes sense to me as well, it seems we should have some interfaces that expose different functionality. In this case, it so happens that the Storage ops are done via JMX, but this could change at some point. I would prefer to keep this name for the class.

Reply:
Thinking more about this, I think I'm mixing the interface that sidecar exposes to the endpoints, and the interface used for the Storage Service in Cassandra. This specific `StorageOperations` is an interface for the REST layer, and this doesn't have to be a mirror of the JMX interface that Cassandra exposes. So I will go ahead and refactor this code to separate the two interfaces.

Reply:
I pushed code to address this comment. Let me know what you think @yifan-c

Reply:
Sounds good to me